framework/src/modules/chain/chain.js

```diff
@@ -171,10 +171,29 @@ module.exports = class Chain {
 				this.options.genesisBlock,
 			);
 
+			this.channel.subscribe('app:state:updated', event => {
+				Object.assign(this.scope.applicationState, event.data);
+			});
+
+			this.logger.info('Modules ready and launched');
+			// After binding, it should immediately load blockchain
+			await this.processor.init(this.options.genesisBlock);
+
+			// Update Application State after processor is initialized
+			this.channel.invoke('app:updateApplicationState', {
+				height: this.blocks.lastBlock.height,
+				lastBlockId: this.blocks.lastBlock.id,
+				maxHeightPrevoted: this.blocks.lastBlock.maxHeightPrevoted || 0,
+				blockVersion: this.blocks.lastBlock.version,
+			});
+
 			// Deactivate broadcast and syncing during snapshotting process
-			if (this.options.loading.rebuildUpToRound) {
+			if (this.options.loading.rebuildUpToRound !== null) {
 				this.options.broadcasts.active = false;
 				this.options.syncing.active = false;
+
+				// Need to reset the BFT to rebuild from start of the chain
+				this.bft.reset();
```

Comment:
this should be done in rebuilder when we do other reset

Reply:
Rebuilder don't have dependency for BFT. Do think we pass this dependency for this line?

Reply:
yes having dependency in the rebuilder is better than splitting the logic in 2 places IMO